### DISCUSSION TITLE
Longhorn manager failure

### DISCUSSION THREAD

**Discussion by usma0118** [longhorn/longhorn#5219] [Q&A] [answered]
Deployment via: Helm chart 1.4.0
Environment: Microk8s

Deployment fails with following error.

`pod/longhorn-manager-nb9cb   Error: failed to generate container "265857fb2654ef89954ce2db9abb741a9ffb0eb68b24a14f779af2e1ff9a5e01" spec: failed to generate spec: path "/var/lib/longhorn/" is mounted on "/var/lib" but it is not a shared mount`

Failed pods/job: 
- longhorn-manager
- longhorn-uninstall job

**Comment by c3y1huang** [comment]
What is your Microk8s version? Looks like mount propagation is not enabled.
Can you provide the result of the [environment check script](https://longhorn.io/docs/1.4.0/deploy/install/#using-the-environment-check-script)?

**Reply by usma0118** [reply]
Seems to be mount propagation.

MicroK8s: v1.25.4 revision 4249
OS: Ubuntu 22.04.1 LTS

environment check is also failing with:
failed to generate spec: path "/tmp/longhorn-environment-check" is mounted on "/tmp" but it is not a shared mount

I have checked open-iscsi is installed.

**Reply by usma0118** [reply]
PS! I am running Ubuntu Core.

**Comment by usma0118** [comment, marked as answer]
Switching from strict snap (confined) to --classic snap fixed the problem.

For reference, I am linking Slack discussion [here](https://kubernetes.slack.com/archives/CAUNWQ85V/p1672996144191339).

**Reply by innobead** [reply]
note: the definition of snap types as a reference for other users. 

https://microk8s.io/docs/install-strict (this will be least privilege, so would have some resource access permission)

https://snapcraft.io/docs/snap-confinement

> **Strict**
> Used by the majority of snaps. Strictly confined snaps run in complete isolation, up to a minimal access level that’s deemed always safe. Consequently, strictly confined snaps can not access your files, network, processes or any other system resource without requesting specific access via an interface ([see below](https://snapcraft.io/docs/snap-confinement#interfaces)).
> 
> **Classic**
> Allows access to your system’s resources in much the same way traditional packages do. To safeguard against abuse, publishing a classic snap requires [manual approval](https://snapcraft.io/docs/reviewing-classic-confinement-snaps), and installation requires the --classic command line argument.
> 
> **Devmode**
> A special mode for snap creators and developers. A devmode snap runs as a strictly confined snap with full access to system resources, and produces debug output to identify unspecified interfaces. Installation requires the --devmode command line argument. Devmode snaps cannot be released to the stable channel, do not appear in search results, and do not automatically refresh.

cc @longhorn/dev
